chapter8/self-modifying-agent/sandbox_runner.py

```
@@ -0,0 +1,161 @@
+        return 2
+    try:
+        payload = json.loads(raw)
+        action = payload["action"]
+        with redirect_stdout(_NullWriter()), redirect_stderr(_NullWriter()):
+            if action == "validate":
+                result = {"checks": _validate(payload)}
+            elif action == "metrics":
+                result = {"metrics": _metrics(payload)}
+            else:
+                return 2
+    except (Exception, SystemExit):
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)